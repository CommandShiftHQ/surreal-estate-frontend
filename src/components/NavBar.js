import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import FacebookLogin from "react-facebook-login";
import "../styles/NavBar.css";

const LogoutButton = ({ onLogout }) => (
  <button type="button" onClick={onLogout}>
    Logout
  </button>
);

const NavBar = ({ onLogin, onLogout, userID }) => (
  <div className="navbar">
    <img
      className="navbar-logo"
      src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
      alt="logo"
    />
    <ul className="navbar-links">
      <li className="navbar-links-item">
        <Link to="/">View Properties</Link>
      </li>
      <li className="navbar-links-item">
        <Link to="/saved-properties">Saved Properties</Link>
      </li>
      <li className="navbar-links-item">
        <Link to="/add-property">Add Property</Link>
      </li>
      {!userID ? (
        <FacebookLogin
          appId="3048483165266445"
          autoLoad
          fields="name,email,picture"
          // onClick={componentClicked}
          callback={onLogin}
        />
      ) : (
        <LogoutButton onLogout={onLogout} />
      )}
    </ul>
  </div>
);

LogoutButton.propTypes = {
  onLogout: PropTypes.func.isRequired,
};

NavBar.propTypes = {
  onLogin: PropTypes.func.isRequired,
  onLogout: PropTypes.func.isRequired,
  userID: PropTypes.string.isRequired,
};

export default NavBar;
