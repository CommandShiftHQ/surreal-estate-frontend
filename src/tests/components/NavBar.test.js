import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import NavBar from "../../components/NavBar";

const renderComponent = () => {
  const { asFragment, getByAltText, getByText } = render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );

  return {
    fragment: asFragment(),
    logo: getByAltText("logo"),
    viewPropertiesLink: getByText("View Properties").closest("a"),
    addPropertyLink: getByText("Add Property").closest("a"),
  };
};

let component;

beforeEach(() => {
  component = renderComponent();
});

test("renders correctly", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

test("it displays a logo", () => {
  expect(component.logo).toBeInTheDocument();
});

test("Properties links to /", () => {
  expect(component.viewPropertiesLink).toHaveAttribute("href", "/");
});

test("Add Property links to /add-property", () => {
  expect(component.addPropertyLink).toHaveAttribute("href", "/add-property");
});
