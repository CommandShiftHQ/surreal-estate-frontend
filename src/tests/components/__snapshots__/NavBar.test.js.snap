// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="logo"
      class="navbar-logo"
      src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
    />
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          href="/"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          href="/saved-properties"
        >
          Saved Properties
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          href="/add-property"
        >
          Add Property
        </a>
      </li>
      <div>
        Facebook login
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
